src/openforms/registrations/contrib/objects_api/tests/test_json_schema.py

```diff
@@ -0,0 +1,319 @@
+    "properties": {
+        "complainant": {"$ref": "#/definitions/person"},
+        "noisyAddress": {"$ref": "#/definitions/address"},
+    },
+    "definitions": {
+        "person": {
+            "type": "object",
+            "properties": {
+                "first.name": {"type": "string"},
+                "last.name": {"type": "string"},
+            },
+        },
+        "address": {
+            "type": "object",
+            "properties": {
+                "street": {"type": "string"},
+            },
+        },
+    },
+}
+
+
+JSON_SCHEMA_NESTED_REFS = {
+    "$id": "noise-complaint.schema",
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "Noise complaint example V2",
+    "type": "object",
+    "additionalProperties": False,
+    "properties": {
+        "complainant": {"$ref": "#/definitions/person"},
+    },
+    "definitions": {
+        "person": {
+            "type": "object",
+            "properties": {
+                "residence": {"$ref": "#/definitions/address"},
+            },
+        },
+        "address": {
+            "type": "object",
+            "properties": {
+                "street": {"type": "string"},
+            },
+        },
+    },
+}
+
+JSON_SCHEMA_UNKOWN_REF = {
+    "$id": "noise-complaint.schema",
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "type": "object",
+    "additionalProperties": False,
+    "properties": {
+        "invalid": {"$ref": "#/invalidref"},
+    },
+}
+
+JSON_SCHEMA_EXTERNAL_REF = {
+    "$id": "noise-complaint.schema",
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "type": "object",
+    "additionalProperties": False,
+    "properties": {
+        "external": {"$ref": "http://example.com/external-schema.json"},
+    },
+}
+
+JSON_SCHEMA_REQUIRED_PATHS = {
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "type": "object",
+    "properties": {
+        "a": {"type": "string"},
+        "b": {
+            "type": "object",
+            "properties": {
+                "c": {"type": "string"},
+                "d": {
+                    "type": "object",
+                    "properties": {
+                        "e": {"type": "string"},
+                        "f": {"type": "string"},
+                    },
+                    "required": ["e"],
+                },
+            },
+            "required": ["c", "d"],
+        },
+    },
+    "required": ["a", "b"],
+}
+
+
+class IterJsonSchemaTests(SimpleTestCase):
+    """Test cases to assert the JSON Schemas are correctly iterated over.
+
+    The first path element being the root one, it is not included as it could
+    simply be tested as ``([], CURRENT_SCHEMA_TESTED)``.
+    """
+
+    def test_iter_json_schema_no_refs(self):
+        paths_list = [
+            (path.segments, schema)
+            for path, schema in iter_json_schema_paths(JSON_SCHEMA_NO_REFS)
+        ]
+
+        self.assertListEqual(
```

Comment:
https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertListEqual

> These methods are used by default when comparing lists or tuples with [assertEqual()](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertEqual).

So in general you should just always be calling `self.assertEqual` and not the type-specific variants.

```suggestion
        self.assertEqual(
```

Reply:
Aaah unittest..